### PR TITLE
Fix/skybox improvements

### PR DESCRIPTION
**Pull request by 0xD-Fabio**
## What does this PR change?

This PR implements general small improvements to the Skybox.
This PR is related also to the Skybox shader PR #4990 

...

## How to test the changes?

1. Launch the explorer
2. See if the skybox works as intended

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### <samp>🤖 Generated by Copilot at fc1a605</samp>

This pull request improves the procedural skybox feature by using an asynchronous method to find and assign the main camera and creating a `SkyboxCamera` object that uses the main camera as a reference. It also refactors the `SkyboxController` and `SkyboxCamera` classes to make them more performant and reliable.

### CODE REVIEW

**Comment by 0xD-Fabio** [comment]
Made a test on the average frame-time at 100,100 coordinates. See the following results:
**renderqueue 2445**
Average frame time: 11.3082 ms
**renderqueue 2000**
Average frame time: 11.5851 ms
**renderqueue 2998**
Average frame time: 12.96443 ms

File: unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/SkyboxCamera.cs [hunk 1]
```diff
@@ -9,6 +9,19 @@ public class SkyboxCamera
         private Camera skyboxCamera;
         private SkyboxCameraBehaviour camBehavior;
 
+        public SkyboxCamera(Camera mainCam)
```
Comment (AjimenezDCL) [inline unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/SkyboxCamera.cs:12]: This whole class is doing stuff like setting the camera to a rendering asset without Post Processing, modifying `renderShadows`, changing the layer and a bunch of super dangerous stuff that's being apply to the main camera. The whole script  shouldnt be needed anymore.

**Review by AjimenezDCL — CHANGES_REQUESTED** [review]

File: unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/SkyboxCamera.cs (hunk 1 above)
Comment (0xD-Fabio) [reply unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/SkyboxCamera.cs:12]: Thanks, I will go more aggressive on that 👍

**Review by AjimenezDCL — APPROVED** [review]

**Review by sandrade-dcl — CHANGES_REQUESTED** [review]

File: unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/SkyboxController.cs [hunk 2]
```diff
@@ -253,10 +236,9 @@ private void AssignCameraInstancetoProbe()
             }
 
             // make probe a child of main camera
-            if (Camera.main != null)
+            if (cameraMainRef != null)
```
[the remark this answers is not in the document]
Comment (0xD-Fabio) [reply unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/SkyboxController.cs:239]: This is a leftover, thanks for spotting it!

[the remark this answers is not in the document]
Comment (sandrade-dcl) [reply unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/SkyboxController.cs:239]: Ok, knowing that you're going to remove it I'm goingt to approve it in order to not to block the PR 👍

**Review by sandrade-dcl — APPROVED** [review]

**Comment by 0xD-Fabio** [comment]
This PR is on hold because QA found a bug with the clipping plane of the camera, due to the last changes.
I wrote more details in the related task #4823

**Comment by 0xD-Fabio** [comment]
I am closing this PR because we need to assess what should be done next to improve the Skybox and the Infinite floor further.
As reported in ticket #4823 these are the final observations.

### Recap and future developments

- The PR with small skybox shader improvements has been merged to dev.

- This PR where the Skybox camera should have been merged to the Main Camera is being closed because the Main Camera has a maximum draw distance of 500. This renders the skybox correctly, but not the infinite floor, which is rendered up until the max distance (set in the option by the user), and is not blending with the skybox (watch attached video).

https://images.zenhubusercontent.com/337227404/15b0b4f1-52eb-47aa-b4ce-c25984db4463/untitled_video___made_with_clipchamp.mp4

### Possible future development:

1. Set the camera draw distance fixed to 3000 (this will permit to draw the infinite floor on the skybox layer up until horizon and blend it correctly with the skybox)

2. Develop a `LayerCullingOverrideController` to change the culling distance for any given layer.

3. Change the "draw distance" option we have in the graphical settings menu to manipulate the culling distance of layer[0] instead of the Camera draw distance of the frustum. (this way, we are setting the draw distance only for the default Layer 0, which contains all the world objects)

The `LayerCullingOverrideController` should be based on this Unity's [Camera.layerCullDistances documented here](https://docs.unity3d.com/ScriptReference/Camera-layerCullDistances.html)
Unity's documentation is not super explicative but adapting their example to our needs:
`distances[0] = 500;`
will set the Default layer 0 to render up to a distance of 500.

_Bonus: with that system in place, we can decide on some granular rendering options on different layers for different graphic levels_

**Pull request closed without merging**